pandas/core/arrays/timedeltas.py

```diff
@@ -231,6 +232,28 @@ def _validate_fill_value(self, fill_value):
                              "Got '{got}'.".format(got=fill_value))
         return fill_value
 
+    def astype(self, dtype, copy=True):
+        # We handle
+        #   --> timedelta64[ns]
+        #   --> timedelta64
+        # DatetimeLikeArrayMixin super call handles other cases
+        dtype = pandas_dtype(dtype)
+
+        if is_timedelta64_dtype(dtype) and not is_timedelta64_ns_dtype(dtype):
+            # essentially this is division
+            result = self._data.astype(dtype, copy=copy)
+            if self._hasnans:
+                values = self._maybe_mask_results(result,
+                                                  fill_value=None,
+                                                  convert='float64')
+                return values
+            return result.astype('i8')
```

Comment:
this doesn't make sense to return i8 here (if above you are returning ``m8``

Reply:
See discussion on the same code in #24024; this is the same behavior we have in master for TimedeltaIndex.

Reply:
is this actually tested? this is not correct

Reply:
This is identical to https://github.com/pandas-dev/pandas/pull/24405/files#diff-0be8361afef6d9bf7222ddb4f8878f36L293. I don't know why that's the behavior on master.

Reply:
This was just moved from TimedeltaIndex to TimedeltaArray